compiler/rustc_incremental/src/persist/fs.rs

```diff
@@ -601,12 +600,39 @@ fn string_to_timestamp(s: &str) -> Result<SystemTime, &'static str> {
     Ok(UNIX_EPOCH + duration)
 }
 
-fn crate_path(sess: &Session, crate_name: Symbol, stable_crate_id: StableCrateId) -> PathBuf {
+fn crate_path(sess: &Session) -> PathBuf {
+    // Use the crate name from the command line if available and valid
+    let crate_name = sess.opts.crate_name.as_ref().and_then(|name| {
+        name.as_str().chars().all(|c| c.is_alphanumeric() || c == '_').then_some(&name[..])
+    });
```

Comment:
does this duplicate some logic from somewhere?

We should probably ensure they stay in sync (ideally by deduplicating, maybe via a `Sess` method?)